// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: player/v1/room.proto

package dbv1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on UserRoomProto with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *UserRoomProto) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on UserRoomProto with the rules defined
// in the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in UserRoomProtoMultiError, or
// nil if none found.
func (m *UserRoomProto) ValidateAll() error {
	return m.validate(true)
}

func (m *UserRoomProto) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Id

	// no validation rules for IsCreator

	if len(errors) > 0 {
		return UserRoomProtoMultiError(errors)
	}

	return nil
}

// UserRoomProtoMultiError is an error wrapping multiple validation errors
// returned by UserRoomProto.ValidateAll() if the designated constraints
// aren't met.
type UserRoomProtoMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m UserRoomProtoMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m UserRoomProtoMultiError) AllErrors() []error { return m }

// UserRoomProtoValidationError is the validation error returned by
// UserRoomProto.Validate if the designated constraints aren't met.
type UserRoomProtoValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e UserRoomProtoValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e UserRoomProtoValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e UserRoomProtoValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e UserRoomProtoValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e UserRoomProtoValidationError) ErrorName() string { return "UserRoomProtoValidationError" }

// Error satisfies the builtin error interface
func (e UserRoomProtoValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sUserRoomProto.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = UserRoomProtoValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = UserRoomProtoValidationError{}
